### ISSUE TITLE
Local DNS Records report different for AA or AAAA records

### ISSUE THREAD

**Issue by eitch** [pi-hole/FTL#925]
### Versions
-  Pi-hole version is v5.1.2 (Latest: v5.1.2)
-  AdminLTE version is v5.1.1 (Latest: v5.1.1)
-  FTL version is v5.2 (Latest: v5.2)

### Platform
- OS and version:  Raspbian GNU/Linux 10 (buster)
- Platform:  Raspberry Pi

### Expected behavior

When i set a local DNS record for an address, then i expect that name to always resolve to the same IP, regardless of type. Clients on my network using e.g. systemd-resolved get different responses, depending on the type of query. This can be tested using dig:

dig @192.168.1.2 AA www.example.com
dig @192.168.1.2 AAAA www.example.com

### Actual behavior / bug

Instead of the local DNS IP defined in the panel, the response for AAAA returns the public IP.

### Steps to reproduce

1. Define a local DNS IP
2. Use dig to check the AA and AAAA records

**Comment by DL6ER** [comment]
There can be no same answer to different query types as the format of the reply depends on the query type itself (since `AA` queries do not exist, I guess you mean `A` queries?).
For instance, `A` queries are IPv4 address queries, they return something like `93.184.216.34`, while `AAAA` queries are IPv6 address queries, returning something like `2606:2800:220:1:248:1893:25c8:1946`.

My assumption now is that you created only one local DNS record (`A` type), like `www.example.com -> 1.2.3.4` and didn't specify an `AAAA` entry. The `AAAA` query returning the publicly available IPv6 address happens because as your configuration does not contain any knowledge of an `AAAA` record for this domain. What else could/should Pi-hole do rather than trying to look up this?

TL;DR This is expected behavior. What you're expecting (`AAAA` local as well) needs a different local configuration.

**Comment by yubiuser** [comment]
Duplicate of

**Comment by eitch** [comment]
How can i create a AAAA record in the pi-hole UI? I only have two fields, the domain name and the IP. But yes you are write, it seems to be because my system has ipv6 enabled.

I have now enabled the ipv6 DNS on the pi-hole, but dig still returns the public IP:

dig @192.168.1.2 AAAA www.example.com


	; <<>> DiG 9.16.6-Ubuntu <<>> @192.168.1.2 AAAA www.example.com
	; (1 server found)
	;; global options: +cmd
	;; Got answer:
	;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 25339
	;; flags: qr rd ra; QUERY: 1, ANSWER: 2, AUTHORITY: 1, ADDITIONAL: 1

	;; OPT PSEUDOSECTION:
	; EDNS: version: 0, flags:; udp: 4096
	; COOKIE: e097702c669174c5 (echoed)
	;; QUESTION SECTION:
	;www.example.com.		IN	AAAA

	;; ANSWER SECTION:
	www.example.com.	1729	IN	CNAME	example.mooo.com.
	example.mooo.com.	1729	IN	CNAME	example.ddns.net.

	;; AUTHORITY SECTION:
	ddns.net.		1729	IN	SOA	nf1.no-ip.com. hostmaster.no-ip.com. 2344634079 10800 1800 604800 1800

	;; Query time: 0 msec
	;; SERVER: 192.168.1.2#53(192.168.1.2)
	;; WHEN: Mon Nov 16 10:21:02 CET 2020
	;; MSG SIZE  rcvd: 230

**Comment by yubiuser** [comment]
> How can i create a AAAA record in the pi-hole UI? 

There is no way to create a AAAA record via UI. You have to modify the config files.
